### PR TITLE
Add AppData file

### PR DESCRIPTION
**Pull request by jurf**
This file allows MuseScore to be seen in apps like GNOME Software and KDE Discover and enables Flatpak packaging.

Note that this file needs edits, as it is basically only a placeholder. It should pass the strict validation:

```
❯ appstream-util validate-strict mscore.appdata.xml 
mscore.appdata.xml: FAILED:
• tag-missing           : <translation> not specified
• value-missing         : <!-- Copyright [year] [name] --> is not present
• tag-invalid           : <updatecontact> should be <update_contact>
• style-invalid         : Content before <ul> is too short [134], at least 300 characters required
• translations-required  : <name> has no translations
• translations-required  : <summary> has no translations
• translations-required  : <description> has no translations
Validation of files failed
```

`<update_contact>` (sorry about the naming error, only just saw it) is set to me for now (since this file is used in Flathub), but I have little knowledge on MuseScore. Translations is something I can’t provide. I’m not sure whom to assign copyright.

Of course you’ll want to update the description to something that sells MuseScore better, and is longer to fit the spec. The screenshot needs replacing also, as for now it’s just a scaled down screenshot from Wikipedia.

[This site](https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html) provides great information on how to construct an appdata file, especially screenshots.

I’d be happy to provide assistance if needed. You can ping me on IRC as `jurf[m]` also.

This PR builds upon the original one (#3115) by @Alexander-Wilms. I’ll add the Flatpak manifest when it successfully builds on all platforms.

### CODE REVIEW

**Comment by jurf** [comment]
BTW MuseScore is now available to [download](https://flathub.org/repo/appstream/org.musescore.MuseScore.flatpakref) from [Flathub](https://flathub.org/apps.html).

**Comment by lasconic** [comment]
Hi, thank you for this! I'm just back from holidays.
To merge this pull request, I need you to sign the MuseScore CLA at https://musescore.org/en/cla Can you do it and let me know what your MuseScore username is?
Thanks!

**Comment by jurf** [comment]
Hey, sorry, gonna have to wait for Monday. Feel free to propose any changes though in the meantime!

**Comment by jurf** [comment]
@lasconic oof, for some reason I completely missed your comment, even though I replied to it.

My MuseScore username is the same as my GitHub one – `jurf`. But please, keep in mind the things I mentioned in the issue description :-)

**Comment by Jojo-Schmitz** [comment]
I believe you should squash the commits into just one

**Comment by jurf** [comment]
Done!

**Comment by jurf** [comment]
OK, pushed a major update, should cover _much_ more things, though the screenshots should be in 16:9 ~1200px wide PNG (preferably on Linux, not macOS like now), not sure if you like the ones I’ve chosen either, and I still have no idea how to do the translations.

**Comment by jurf** [comment]
Anyway, after this update I think it is ready to be merged.

**Comment by jurf** [comment]
Done.

**Comment by jurf** [comment]
I also merged the current version used on Flathub, you can see it in action [here](https://flathub.org/apps/details/org.musescore.MuseScore).

**Comment by jurf** [comment]
OK, this is getting slightly frustrating. It’ll soon be a _year_ since I opened this PR, and it’s not even a code-related one. I’d really appreciated if someone merged this.

**Comment by anatoly-os** [comment]
@jurf sure. I'll double check tomorrow and merge.

**Comment by jurf** [comment]
Thanks.

File: build/Linux+BSD/mscore.appdata.xml.in [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+    </p>
+    <p>
+      MuseScore includes a set of sounds that reproduce common instruments (as defined by General MIDI) without taking up a lot of disk space or memory, but you can also substitute any SoundFont you prefer for a wider variety of sounds or for more realism.
+    </p>
+    <p>
+      MuseScore can import and export MIDI and MusicXML files, and it can also import from Capella and several other programs. MuseScore can export to PDF, PNG, and other graphic formats, to WAV and other audio formats, or to Lilypond for an alternative layout and print option.
+    </p>
+  </description>
+
+  <categories>
```
Comment (anatoly-os) [inline build/Linux+BSD/mscore.appdata.xml.in]: I would add SheetMusic, NotationSoftware categories.

File: build/Linux+BSD/mscore.appdata.xml.in [hunk 2]
```diff
@@ -0,0 +1,316 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2017 Juraj Fiala <jurf_AT_riseup.net> -->
```
Comment (anatoly-os) [inline build/Linux+BSD/mscore.appdata.xml.in]: According to CLA, we have to put Werner Schweer name to any copyright mentions. You can replace this with text like
`Copyright 2018 Werner Schweer and others. Maintainer of the file is Juraj Fiala <jurf_AT_riseup.net>.

**Comment by anatoly-os** [comment]
I reviewed the appdata file. I'm not able to check it on Unix machine, so I believe you did.
I have questions, btw. I didn't get the purpose of <releases> tag. It doesn't include the 3.0 version which is clear, but the PR is for master, where 3.0 release is scheduled. How should we support 3.0 Release in this file in future?
Should we change the file before the official release or it can be changed few days after by contributors?

File: build/Linux+BSD/mscore.appdata.xml.in (hunk 2 above)
Comment (jurf) [reply build/Linux+BSD/mscore.appdata.xml.in]: Sure, will do.

File: build/Linux+BSD/mscore.appdata.xml.in (hunk 1 above)
Comment (jurf) [reply build/Linux+BSD/mscore.appdata.xml.in]: I just copied them over from the desktop file (now that I think of it I need to check whether I _should_ have done that, I have a feeling that should be done automatically).

**Comment by jurf** [comment]
Yeah, thanks for bringing this up, I completely forgot that this is master.

It’s supposed to be added _before_ release, so that release notes and version info are shown in app centres (e.g. GNOME Software), which is the whole purpose of that section. If you don’t have the release notes done at release time then that’s a problem, in that case I’d have to patch it in downstream.

I’ll add at least a 3.0 tag into it, release notes can be added later, when they’re available.

Thanks for your interest. :slightly_smiling_face:

File: build/Linux+BSD/mscore.appdata.xml.in (hunk 1 above)
Comment (jurf) [reply build/Linux+BSD/mscore.appdata.xml.in]: Yeah, it’s pulled from the desktop file, so this is a different issue.

**Comment by jurf** [comment]
OK, all done.

**Comment by anatoly-os** [comment]
@jurf thank you for the appData file! Filling this file with actual release notes is one more requirement we have to meet, but I hope it will worth it.

**Comment by jurf** [comment]
Thanks for the merge!

Though lol at the timing. :grinning: I just merged some changes to the AppData used in Flathub the day before yesterday, and didn’t yet have the time to merge them here. Is it alright if I create a new PR? They’re only minor.

**Comment by anatoly-os** [comment]
Sure, you're welcome. Just create a new PR with the changes.

**Pull request merged**